reedstrm/baking_workflow_and_errors.md

```diff
@@ -0,0 +1,120 @@
+
+When a new version of a book is published, we will attempt to bake it with the recipe specified by the `print_style` parameter. There will be a table of recipes that are the _current_ recipe for given print style. If baking with that recipe fails, we will then retry baking with the last successful recipe for this book. If that succeeds, we update the webview, but also mark the state as `stale_recipe`. This way, the new content is available, even if changes in presentation from a new recipe are not. Note that uploading a new default recipe for a given `print_style` will also put books in this state.
+
+If the `print_style` parameter value does not have a default recipe associated with it, we will then check to see if a file with that name has been attached to the book. If there is no file, the book will be marked `Done/Success`. This will allow us to rollout recipes gradually for existing print styles. If there is a file, this will serve as a local recipe for baking a specific book, while removing the need for a "magic filename" from our processing code. The fallback described above would still be used - if the current book doesn't bake, use the most recent successful recipe, unless it is the same one we just tried.
+
+## Content Fallback and Delayed Baking
+
+If both attempts to bake fail, we leave the existing baked content in place, and mark the book as `stale_content`. The latest version of the content will still be available, by viewing the raw version, via the `?is_baked=false` query parameter.
+
+Publishing individual pages causes minor-version revisions of each book that contains that page. This leads to a thundering-herd problem when trying to do several page edits in a row each start a book, and all of its derived copies, to bake. To help resolve this, we will no longer trigger baking on minor version revisions. We instead will mark these books as
+`pending`, and process them on a delayed schedule. An interface for directly triggering baking for a given book in the pending state will be provided, but a time-based process will also "sweep up" books in this state. However, only the most recent minor version will be baked - when multiple pages and edited and published, each publish event will set the book with the new minor version to `pending`, while setting the previously pending minor version to `raw_only`.
+
+## Reporting Baking State
+
+The state of baking of a given book will be available in the `/extras` page for that book. This will be used to drive additional UI in the `More Information` tab, where the "Print Style" is currently reported. If the state is `pending` or `stale_recipe`, a link to trigger immediate baking will be provided (this is a `POST`). Also, the most recent history of baking, with error messages, will be available here.
```

Comment:
:nit: I'm not sure that a link can perform a `POST`. It is good to have it as a `POST` because crawlers ~~will~~ should not try to use it.

RE the recent history... could that be a link to another page? I don't think the "build history" needs a UX polish (but I'm OK if there is one)